pkg/ceph/osd/daemon.go

```diff
@@ -38,6 +38,11 @@ const (
 )
 
 func Run(context *clusterd.Context, agent *OsdAgent) error {
+	if err := setHostName(context); err != nil {
```

Comment:
Agree. NodeName is available as a downward API as of 1.4. https://github.com/kubernetes/kubernetes/pull/27880